rag-and-text-to-sql/rag_code.py

```
@@ -0,0 +1,364 @@
+                'nullable': column['nullable']
+            }
+        
+        return schema_info
+
+class OllamaLLM:
+    """Custom LLM Wrapper for Ollama"""
+
+    def __init__(self, model="Meta-Llama-3.1-8B-Instruct"):
+        self.client = OpenAI(
+            base_url="https://api.sambanova.ai/v1",
+            api_key=api_key
```

Ruff (0.8.2)

Undefined name `api_key`

(F821)